### ISSUE TITLE
[bug] Plugin does not work in jmeter server/distributed mode

### ISSUE THREAD

**Issue by georgejdli** [rollno748/Extended-csv-dataset-config#19]
(Thanks for creating and supporting this plugin!)

## How to replicate: run jmeter in distributed mode
- command to run on the worker: `bin/jmeter-server -Dserver.rmi.localport=50000 -Dserver_port=1099`

### The error message
- this error message was only visible in 5.6.3 (5.6.2 did not properly display the error)
```log
2024-03-05 22:48:01,419 INFO c.d.j.u.FileServerExtended: Default base='/jmeter/apache-jmeter-5.6.3/bin'
2024-03-05 22:48:01,433 ERROR o.a.j.e.StandardJMeterEngine: Unable to execute testStarted(10.240.0.158) for test element com.di.jmeter.config.ExtendedCsvDataSetConfig@b005ddcc
java.lang.NullPointerException: null
    at com.di.jmeter.config.ExtendedCsvDataSetConfig.testStarted(ExtendedCsvDataSetConfig.java:210) ~[di-extended-csv-2.1.jar:?]
    at com.di.jmeter.config.ExtendedCsvDataSetConfig.testStarted(ExtendedCsvDataSetConfig.java:227) ~[di-extended-csv-2.1.jar:?]
    at org.apache.jmeter.engine.StandardJMeterEngine.notifyTestListenersOfStart(StandardJMeterEngine.java:249) [ApacheJMeter_core.jar:5.6.3]
    at org.apache.jmeter.engine.StandardJMeterEngine.run(StandardJMeterEngine.java:432) [ApacheJMeter_core.jar:5.6.3]
    at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
    at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
    at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
    at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
    at java.base/java.lang.Thread.run(Thread.java:829) [?:?]
```

### source code
- looking at the source code I can see a check for isNonGUI
  - https://github.com/rollno748/Extended-csv-dataset-config/blob/0f840d0f5a85f4f0923c8d09180bf8b1567f310d/src/main/java/com/di/jmeter/config/ExtendedCsvDataSetConfig.java#L206
- looking at the jmeter source code for that function (in v5.1.1 as configured in this plugin's POM.xml
```java
    public static boolean isNonGUI() {
      return "true".equals(System.getProperty("JMeter.NonGui"));
   }
```
- doing a check on the property value with a JSR223 sampler we see that JMeter.NonGui: null
  - hence the NullPointerException error with the plug in
```
log.info("JMeter.NonGui: {}", System.getProperty("JMeter.NonGui"))

2024-03-05 23:23:16,510 INFO o.a.j.p.j.s.J.JSR223 Sampler: JMeter.NonGui: null
```

**Comment by rollno748** [comment]
Hey @georgejdli 

Thanks for bringing up this issue.
I have never tested this on distributed testing. Will give it a try and fix it at the earliest

Thanks

**Comment by georgejdli** [comment]
I'm not sure if there's a more robust way to determine if Jmeter is running in server mode but this is my quick fix to get the plugin working in distributed mode. The system property `server_port` will always be set for my use case so I do a check for that.

```java
private boolean isServerMode() {
        return System.getProperty("server_port") != null;
    }

    @Override
    public void testStarted() {
        FileServerExtended fileServer = FileServerExtended.getFileServer();
        if(JMeter.isNonGUI() || isServerMode()){
            String baseDirectory = org.apache.jmeter.services.FileServer.getFileServer().getBaseDir();
            fileServer.setBasedir(baseDirectory);
        }else {
            String testPlanFile = GuiPackage.getInstance().getTestPlanFile();
            fileServer.setBasedir(testPlanFile);
        }
    }
```

**Comment by rollno748** [comment]
Hey that is what I thought of doing it - Its like i am re-inventing the wheel  :P
Can you just do a Pull request ?
contributions are always welcome mate !

**Comment by georgejdli** [comment]
> Hey that is what I thought of doing it - Its like i am re-inventing the wheel :P Can you just do a Pull request ? contributions are always welcome mate !

I have opened a PR

**Comment by rollno748** [comment]
Apologies, I was busy away...
Thanks for the PR - I will merge it tomorrow

Thanks again

**Comment by rollno748** [comment]
Fixes were merged to master and release has been published to plugins manager (2.2)
Thanks @georgejdli for the contribution

Do support us by hitting star and feel free to open up issues if you experience a bug 
If you have ideas - please feel free to create a feature request :)
